nemo_curator/modules/distributed_data_classifier.py

```diff
@@ -96,6 +150,126 @@ def _filter_documents(
         raise TypeError("filter_by must be a string or list type")
 
 
+def _run_classifier_helper(
+    df: "dask_cudf.DataFrame",
+    model: "HFModel",
+    labels: list[str],
+    max_chars: int,
+    batch_size: int,
+    label_col: str,
+    prob_col: str = None,
+    keep_prob: bool = False,
+) -> "dask_cudf.DataFrame":
+
+    if keep_prob and prob_col is None:
+        raise ValueError("prob_col must be provided if keep_prob is True")
+
+    if prob_col is not None:
+        keep_prob = True
+
+    prob_internal_col = "_prob"
+    # TODO: Make crossfit handle this cleanly
+    pred_internal_col = "labels"
+
+    df["sliced_text"] = df["text"].str.slice(0, max_chars)
+    columns_to_keep_list = df.columns.to_list()
+    columns_to_keep_list.remove("sliced_text")
+
+    classifier_pipe = op.Sequential(
+        op.Tokenizer(model, cols=["sliced_text"], tokenizer_type="sentencepiece"),
+        op.Predictor(
+            model,
+            sorted_data_loader=True,
+            batch_size=batch_size,
+            pred_output_col=prob_internal_col,
+        ),
+        repartition=df.npartitions,
+        keep_cols=columns_to_keep_list,
+    )
+    df = classifier_pipe(df)
+    # TODO: Make crossfit handle this cleanly
+    # to prevent the labeler from dropping the prob_internal_col
+    # and combine it into a single step
+    labeling_pipe = op.Sequential(
+        op.Labeler(labels, cols=[prob_internal_col]),
+        keep_cols=columns_to_keep_list + [prob_internal_col],
+    )
+    df = labeling_pipe(df)
+    if keep_prob:
+        df = df.rename(
+            columns={pred_internal_col: label_col, prob_internal_col: prob_col}
+        )
+    else:
+        df = df.rename(columns={pred_internal_col: label_col})
+        df = df.drop(columns=[prob_internal_col])
+
+    return df
+
+
+class DomainModel(HFModel):
+    def __init__(self, config, out_dim=None, model_path=None, autocast=False):
+        self.config = config
+        self.out_dim = out_dim
+        self.model_path = model_path
+        self.autocast = autocast
+        super().__init__(self.config.model)
+
+    def load_model(self, device="cuda"):
+        model = CustomModel(
+            self.config,
+            out_dim=self.out_dim,
+            config_path=None,
+            pretrained=True,
+            autocast=self.autocast,
+        )
+        model = model.to(device)
+        if os.path.exists(self.model_path):
```

Comment:
Given that the models are going up on HuggingFace soon too, would it make sense to add an additional check after the local disk check to see if the `model_path` is valid on HF and if so get it from there?

Reply:
Lets do that in a followup .

Reply:
To track: https://github.com/NVIDIA/NeMo-Curator/issues/72